File: ruby-rails/lib/generators/rerout/templates/rerout.rb
# frozen_string_literal: true

# Rerout configuration.
#
# Keep real credentials out of source control — prefer Rails encrypted
# credentials (`Rails.application.credentials`) or environment variables.
#
# Docs: https://rerout.co/docs
Rerout::Rails.configure do |config|
  # Project API key (`rrk_…`). Required to use `Rerout::Rails.client`.
  config.api_key = ENV.fetch('REROUT_API_KEY', nil)

  # Endpoint signing secret (`whsec_…`). Required for webhook verification.
  config.webhook_secret = ENV.fetch('REROUT_WEBHOOK_SECRET', nil)

  # Override the API base URL (staging / self-hosted). Optional.
  # config.base_url = 'https://api.rerout.co'

  # Per-request timeout in seconds. Optional, defaults to 30.
  # config.timeout = 30

  # Webhook signature timestamp tolerance in seconds. `0` disables the
  # staleness check. Optional, defaults to 300.
  # config.signature_tolerance_seconds = 300
end

# React to verified webhook deliveries via ActiveSupport::Notifications.
# `rerout.webhook` fires for every delivery; per-event topics
# (`rerout.link.clicked`, `rerout.qr.scanned`, …) fire for known events.
#
# ActiveSupport::Notifications.subscribe('rerout.link.clicked') do |event|
#   code = event.payload[:body]['code']
#   Rails.logger.info("Rerout: link #{code} clicked")
# end

File: php-laravel/config/rerout.php
<?php

declare(strict_types=1);

return [

    /*
    |--------------------------------------------------------------------------
    | API Key
    |--------------------------------------------------------------------------
    |
    | The project API key (`rrk_…`) used to authenticate every request to the
    | Rerout API. Generate one in the Rerout dashboard. Required — the bound
    | client throws on construction if this is blank.
    |
    */

    'api_key' => env('REROUT_API_KEY'),

    /*
    |--------------------------------------------------------------------------
    | Base URL
    |--------------------------------------------------------------------------
    |
    | The Rerout API base URL. Leave null to use the production default
    | (https://api.rerout.co). Override for staging or self-hosted setups.
    |
    */

    'base_url' => env('REROUT_BASE_URL'),

    /*
    |--------------------------------------------------------------------------
    | Request Timeout
    |--------------------------------------------------------------------------
    |
    | Per-request timeout in seconds applied to every Rerout API call.
    |
    */

    'timeout' => (int) env('REROUT_TIMEOUT', 30),

    /*
    |--------------------------------------------------------------------------
    | Webhook
    |--------------------------------------------------------------------------
    |
    | Configuration for the bundled webhook controller.
    |
    | - secret:    Endpoint signing secret (`whsec_…`) used to verify the
    |              `X-Rerout-Signature` header on inbound deliveries.
    | - tolerance: Timestamp staleness window in seconds. Set 0 to disable the
    |              replay-protection check.
    | - route:     When `enabled` is true the package registers a POST route at
    |              `path` that runs the bundled WebhookController. Disable it to
    |              wire the controller into your own routes instead.
    |
    */

    'webhook' => [
        'secret' => env('REROUT_WEBHOOK_SECRET'),

        'tolerance' => (int) env('REROUT_WEBHOOK_TOLERANCE', 300),

        'route' => [
            'enabled' => (bool) env('REROUT_WEBHOOK_ROUTE_ENABLED', true),
            'path' => env('REROUT_WEBHOOK_ROUTE_PATH', 'rerout/webhook'),
            'middleware' => ['api'],
        ],
    ],

];
